Std/Data/Bool.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2023 F. G. Dorais. No rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: F. G. Dorais
+-/
+
+import Std.Tactic.Alias
+
+/-- Boolean exclusive or -/
+abbrev xor : Bool → Bool → Bool := bne
+
+@[inherit_doc] infixl:30 " ^^ " => xor
```

Comment:
I'm not entirely sure about making this a global notation. It's a natural companion to the other `Bool` operators `!`, `&&`, `||`.

Reply:
Should we reuse `Xor`? It has a correct type signature in this case.

Reply:
I don't think we should have xor at all. If mathlib wants it, okay, but it's just a synonym for an existing definition with a notation already.

Reply:
The abbreviation is still useful since `bne` is polymorphic. I moved `xor` in the `Bool` namespace and removed the notation. Mathlib can add `export Bool (xor)` somewhere, if necessary.